### ISSUE TITLE
Can not collect filtered string features information

### ISSUE THREAD

**Issue by dixingxing0** [DeepRec-AI/DeepRec#944]
**System information**
- DeepRec version or commit id: 2306

**Describe the current behavior**
When the feature is of string type, if the number of filtered feature keys is 0, and when the feature is of int64 type, you can correctly obtain the information of the filtered feature.

**Describe the expected behavior**
Here is the output for 64 type filtered features:
![image](https://github.com/DeepRec-AI/DeepRec/assets/1303530/3df52363-adc5-4139-8d3b-8130d92f4286)

**Code to reproduce the issue**

Filter definition:
```python
    evict_opt = tf.GlobalStepEvict(steps_to_live=4000)
    filter_option = tf.CounterFilter(filter_freq=3)
    
    ev_opt = tf.EmbeddingVariableOption( evict_option=evict_opt, filter_option=filter_option)
    categorical_column = tf.feature_column.categorical_column_with_embedding(col,
                                                                                 dtype=tf.string,
                                                                                 ev_option=ev_opt)
```
code for collect information:
```python
import tensorflow as tf
from tensorflow.core.protobuf.meta_graph_pb2 import MetaGraphDef
from tensorflow.python.framework import meta_graph
ckpt_path = '../sug_ple_model/'
path = ckpt_path + '/model.ckpt-13359.meta'
meta_graph = meta_graph.read_meta_graph_file(path)
ev_node_list = []
for node in meta_graph.graph_def.node:
    if node.op == 'KvVarHandleOp':
        ev_node_list.append(node.name)
print("ev node list", ev_node_list)
# filter ev-slot
non_slot_ev_list = []
for node in ev_node_list:
    if "Adagrad" not in node:
        non_slot_ev_list.append(node)
print("ev (exculde slot) node list", non_slot_ev_list)
for name in non_slot_ev_list:
    variable_keys = tf.train.load_variable(ckpt_path, name + '-keys')
    print("已准入特征的id个数:%d" % len(variable_keys))
    variable_values = tf.train.load_variable(ckpt_path, name + '-values')
    # print(name + '-values len', len(variable_values))
    # print(name + '-values', variable_values[0])
    variable_freqs = tf.train.load_variable(ckpt_path, name + '-freqs')
    # print(name + '-freqs len', len(variable_freqs))
    # print(name + '-freqs', variable_freqs[0])
    variable_versions = tf.train.load_variable(ckpt_path, name + '-versions')
    # print(name + '-versions', variable_versions[0])
    variable_keys_filtered = tf.train.load_variable(ckpt_path, name + '-keys_filtered')
    print("未准入的特征id个数：%d" % len(variable_keys_filtered))
```

Provide a reproducible test case that is the bare minimum necessary to generate the problem.

**Other info / logs**

**Comment by candyzone** [comment]
Thanks for your issue and we will check it.

**Comment by candyzone** [comment]
it works in my testcase.

import tensorflow as tf
import os

checkpoint_directory = '/tmp/1'
with tf.device("/cpu:0"):
  columns = tf.feature_column.categorical_column_with_embedding("col_emb", dtype=tf.string,
                                                        ev_option = tf.EmbeddingVariableOption(filter_option=tf.CounterFilter(filter_freq=3)))
  W = tf.feature_column.embedding_column(columns,
          dimension=3,
          initializer=tf.ones_initializer(tf.float32))
  ids={}
  ids["col_emb"] = tf.SparseTensor(indices=[[0,0],[1,0],[2,0],[3,0],[4,0],[5,0]], values=['1','1','1','2','2','3'], dense_shape=[6, 1])
  emb = tf.feature_column.input_layer(ids, [W])
fun = tf.multiply(emb, 2.0, name='multiply')
loss = tf.reduce_sum(fun, name='reduce_sum')
gs = tf.train.get_or_create_global_step()
opt = tf.train.AdagradOptimizer(0.1)
g_v = opt.compute_gradients(loss)
train_op = opt.apply_gradients(g_v, gs)
saver = tf.train.Saver()
init = tf.global_variables_initializer()
model_path = os.path.join(checkpoint_directory,
                          "model1.ckpt")
with tf.Session() as sess:
  sess.run([init])
  sess.run([train_op])
  save_path = saver.save(sess, model_path)
  for name, shape in tf.train.list_variables(model_path):
    print(name, shape)

**Comment by dixingxing0** [comment]
We discovered that our string column wasn't a SparseTensor. After making a minor adjustment, it functioned as expected.   Thanks @candyzone .
![image](https://github.com/DeepRec-AI/DeepRec/assets/1303530/d92fee85-9186-4a3a-b111-19c155580612)
